ansible/roles/flannel/tasks/firewalld.yml

```diff
@@ -0,0 +1,5 @@
+---
+# https://bugzilla.redhat.com/show_bug.cgi?id=1033606 and I think others say firewalld+docker == bad
+# It is also disabled by the node role.
+- name: disable firewalld
+  service: name=firewalld enabled=no state=stopped
```

Comment:
Okay, so within openshift-ansible we disabled firewalld across the board, mainly as a way to ensure that it is disabled before we install, start and configure iptables. The tricky part being that iptables doesn't provide a native or supported way to atomically update and persist firewall rules. It looks like the --wait flag was added as a workaround, but it seems like we still have plenty of room for race conditions with updating the firewall rules.

We sort of worked around this by only updating a single chain, to limit the potential for rule change collisions to happen only when adding the jump rule to our new chain.

With that background out of the way...  openshift-sdn currently manages adding it's own port to the firewall (https://github.com/openshift/openshift-sdn/blob/1694e9adbaae43a42944de1c5243a12dca46a535/ovssubnet/controller/kube/bin/openshift-sdn-kube-subnet-setup.sh for example), so we aren't currently managing those rules through openshift-ansible.

As for detecting docker running, I believe there is also code for that in openshift-sdn when setting up the bridge, but @rajatchopra would be able to better speak to that.

I'd say the more you can avoid messing with iptables rules in ansible the better.